### ISSUE TITLE
Chronos: Different batch sizes are used implicitly in Forecasters on different machines

### ISSUE THREAD

**Issue by yushan111** [intel-analytics/BigDL#4705]
### Issue statement
Since Chronos automatedly trains with nano multi-processing, the total `batch_size` varies with the core number on the users' machine. It may result in unreproducible fitting results when users try running the same code on different machines. Besides, since the total `batch_size` varies, while learning rate stays the same, it may also harm the convergence. 

### Detailed explanation
Chronos Forecaster automatedly runs multi-processing, and the `num_processes` depends on the machine core number ( `num_processes = max(1, current_num_threads//8`)). And each process trains `batch_size` samples per iteration, in which case `total_batch_size` adds up to `num_processes * batch_size`.

So if users run 
```python
forecaster = Forecaster(batch_size=32, lr=0.001)
```
on different machines, say machine 1 with 8 cores and machine 2 with 40 cores, there will be a 5x difference in `batch_size`. However, since the learning rate stays the same, users may get completely different fit results,  and may even have problems with convergence.

### Possible fix and discussion
In mutli-processing, the actual `batch_size` in each process = user input `batch_size` / `num_processes`. However, I am not sure about the acceleration with multi-processing in nano vs the stock pytorch in that way, e.g 2 processes each with `batch_size = 16` in nano vs stock pytorch with `batch_size=32`

**Comment by TheaperDeng** [comment]
fixed in https://github.com/intel-analytics/BigDL/pull/4480, will be merged later
